istioctl/pkg/precheck/precheck.go

```diff
@@ -122,10 +118,126 @@ See %s for more information about causes and resolutions.`, url.ConfigAnalysis)
 		fmt.Sprintf("The severity level of precheck at which to display messages. Valid values: %v", diag.GetAllLevelStrings()))
 	cmd.PersistentFlags().StringVarP(&msgOutputFormat, "output", "o", formatting.LogFormat,
 		fmt.Sprintf("Output format: one of %v", formatting.MsgOutputFormatKeys))
+	cmd.PersistentFlags().StringVarP(&fromCompatibilityVersion, "from-version", "f", "",
+		"check changes since the provided version")
 	opts.AttachControlPlaneFlags(cmd)
 	return cmd
 }
 
+func checkFromVersion(ctx cli.Context, revision, version string) (diag.Messages, error) {
+	cli, err := ctx.CLIClientWithRevision(revision)
+	if err != nil {
+		return nil, err
+	}
+	major, minors, ok := strings.Cut(version, ".")
+	if !ok {
+		return nil, fmt.Errorf("invalid version %v, expected format like '1.0'", version)
+	}
+	if major != "1" {
+		return nil, fmt.Errorf("expected major version 1, got %v", version)
+	}
+	minor, err := strconv.Atoi(minors)
+	if err != nil {
+		return nil, fmt.Errorf("minor version is not a number: %v", minors)
+	}
+
+	var messages diag.Messages = make([]diag.Message, 0)
+	if minor <= 20 {
+		// VERIFY_CERTIFICATE_AT_CLIENT and ENABLE_AUTO_SNI
+		if err := checkDestinationRuleTLS(cli, &messages); err != nil {
+			return nil, err
+		}
+		// ENABLE_EXTERNAL_NAME_ALIAS
+		if err := checkExternalNameAlias(cli, &messages); err != nil {
+			return nil, err
+		}
+		// PERSIST_OLDEST_FIRST_HEURISTIC_FOR_VIRTUAL_SERVICE_HOST_MATCHING
+		// TODO
+		messages.Add(msg.NewUnknownUpgradeCompatibility(nil,
```

Comment:
Want me to implement this in a follow up?

Reply:
Yeah if possible would be nice. We don't _have_ to add checks for everything, but it can reduce false positives for users

Reply:
Yeah I can check for duplicate wildcards fairly easily. I'll add it to my TODOs